client/quantum_serverless/core/job.py

```diff
@@ -203,21 +215,34 @@ def run(  # pylint: disable=too-many-locals
                 )
 
             with open(artifact_file_path, "rb") as file:
+                data = {
+                    "title": program.title,
+                    "entrypoint": program.entrypoint,
+                    "arguments": json.dumps(arguments or {}, cls=QiskitObjectsEncoder),
+                    "dependencies": json.dumps(program.dependencies or []),
+                }
+                if config:
+                    config_string = json.dumps(
+                        {
+                            "workers": config.workers,
```

Comment:
dataclasses support `asdict` function.
you can do something like `json.dumps(asdict(config))